test/s3select_test.cpp

```diff
@@ -2603,3 +2661,47 @@ TEST(TestS3selectFunctions, nested_query_multirow_result)
   s3select_res = run_s3select(input_query, input_csv);
   EXPECT_EQ(s3select_res, expected_res);
 }
+
+TEST(TestS3selectFunctions, opserialization_expressions)
+{
+  std::string input;
+  size_t size = 10;
+  generate_rand_columns_csv(input, size);
+
+  char a[5] = {'@', '#', '$', '%'};
+  char b[4] = {'!', '^', '&', '*'};
+  char x = a[rand() % 4];
+  char y = b[rand() % 4];
+
+  const std::string input_query = "select * from s3object ;";
+
+  run_s3select_test_opserialization(input_query, input, &x, &y);
+
+  const std::string input_query_1 = "select int(_1), int(_2) from s3object where  nullif(_1, _2) is not null;";
+
+  std::string s3select_result_1 = run_s3select_opserialization_quot(input_query_1,input, true);
+
+  const std::string input_query_2 = "select int(_1), int(_2) from s3object where int(_1) != int(_2);";
+
+  std::string s3select_result_2 = run_s3select_opserialization_quot(input_query_2,input, true);
+
+  ASSERT_EQ(s3select_result_1, s3select_result_2);
+
+  const std::string input_query_3 = "select int(_1), int(_2) from s3object where int(_1) != int(_2);";
+
+  std::string s3select_result_3 = run_s3select_opserialization_quot(input_query_3,input);
+
+  ASSERT_NE(s3select_result_1, s3select_result_3);
+
+  const std::string input_query_4 = "select int(_1), int(_2) from s3object where  nullif(_1, _2) is not null;";
+
+  std::string s3select_result_4 = run_s3select_opserialization_quot(input_query_4,input, true, x);
+
+  const std::string input_query_5 = "select int(_1), int(_2) from s3object where int(_1) != int(_2);";
+
+  std::string s3select_result_5 = run_s3select_opserialization_quot(input_query_5,input, true, x);
+
+  ASSERT_EQ(s3select_result_4, s3select_result_5);
+
+  ASSERT_NE(s3select_result_4, s3select_result_1);
+}
```

Comment:
the test does create multiple rows/multiple columns results(on different queries).
but it does not validate the results themselves, the output serialization may produce wrong results for both queries.
(no need for several tests which does the same)

its possible to use **constant** input (for multiple rows/columns)
such as bellow (make you own input) 
and validate results for asneeded / always (it different results).
validate the quote and escape 

12,45,"single column,with,delimiteres",64,00
89,88,column with \\"escape quot\\",938,99
12,888,column with space only,88, 11